services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -6724,42 +6717,69 @@ void checkResourceCleanupBeforeStoreCreation(String clusterName, String storeNam
       throw new VeniceException("Store: " + storeName + " still exists in cluster: " + clusterName);
     }
 
-    Set<String> allRelevantStores = new HashSet<>();
-    Arrays.stream(VeniceSystemStoreType.values()).forEach(s -> allRelevantStores.add(s.getSystemStoreName(storeName)));
-    allRelevantStores.add(storeName);
-
-    // Check all the topics belonging to this store.
-    Set<String> topics = getTopicManager().listTopics();
     /**
      * All store version topics from before deletion will not be used since we will create a new version number
      * that has not been used previously. We let TopicCleanupService take care of the cleanups.
      *
      * The version check skip was introduced when Venice was running with Kafka MirrorMaker to avoid a KMM crash.
      * TODO: Evaluate if this code needs to change now that KMM logic has been deprecated.
      *
-     * So for topic check, we will ensure the RT topic for the Venice store will be deleted, and all other system
-     * store topics.
+     * So for topic check, we ensure RT topics for the Venice store and its system stores are deleted.
      */
+    checkKafkaTopicAndHelixResource(clusterName, storeName, false, checkHelixResource, false);
+  }
+
+  void checkKafkaTopicAndHelixResource(
+      String clusterName,
+      String storeName,
+      boolean checkVersionTopic,
+      boolean checkHelixResource,
+      boolean checkOfflinePush) {
+    Set<String> allRelevantStores = new HashSet<>();
+    Arrays.stream(VeniceSystemStoreType.values())
+        .filter(VeniceSystemStoreType::isStoreZkShared)
+        .forEach(s -> allRelevantStores.add(s.getSystemStoreName(storeName)));
```

Comment:
We also need the isStoreZkShared filter. Otherwise, BATCH_JOB_HEARTBEAT, which is not a per user store system store, will be added into the set. So I changed it to:
```
Set<String> allRelevantStores = Arrays.stream(VeniceSystemStoreType.values()).filter(VeniceSystemStoreType::isStoreZkShared).map(s -> s.getSystemStoreName(storeName)).collect(Collectors.toSet());
```

Reply:
Yes. That's right. I missed it. My main point was about replacing `new HashSet` and `foreach` with native stream ops